### PR TITLE
API create phieu dieu tri

### PR DESCRIPTION
**Pull request by hungbg93**
API create phieu dieu tri

### CODE REVIEW

File: app/Repositories/PhieuChamSoc/PhieuChamSocRepository.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+use App\Models\PhieuChamSoc;
+
+class PhieuChamSocRepository extends BaseRepositoryV2
+{
+    public function getModel()
+    {
+        return PhieuChamSoc::class;
+    }
+    
+    function createPhieuChamSoc(array $input)
```
Comment (thien-dang-robustaeng) [inline app/Repositories/PhieuChamSoc/PhieuChamSocRepository.php]: createPhieuChamSoc->create

File: app/Repositories/PhieuChamSoc/PhieuChamSocRepository.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    {
+        $id = $this->model->create($input)->id;
+        return $id;
+    }
+    
+    function getPhieuChamSocById($id)
```
Comment (thien-dang-robustaeng) [inline app/Repositories/PhieuChamSoc/PhieuChamSocRepository.php]: getPhieuChamSocById->getById

File: app/Repositories/PhieuChamSoc/PhieuChamSocRepository.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    {
+        $data = $this->model->findOrFail($id);
+        return $data;
+    }    
+    
+    function getListPhieuChamSocByHsbaId($hsbaId)
```
Comment (thien-dang-robustaeng) [inline app/Repositories/PhieuChamSoc/PhieuChamSocRepository.php]: getListPhieuChamSocByHsbaId->getListByHsbaId

File: app/Services/DieuTriService.php [hunk 4]
```diff
@@ -446,4 +446,21 @@ public function getPhieuDieuTri(array $input)
         $result = $this->dieuTriRepository->getPhieuDieuTri($input);
         return $result;
     }
+    
+    public function createPhieuDieuTri($input)
```
Comment (thien-dang-robustaeng) [inline app/Services/DieuTriService.php]: createPhieuDieuTri->create

File: app/Services/PhieuChamSocService.php [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+use App\Repositories\PhieuChamSoc\PhieuChamSocRepository;
+
+class PhieuChamSocService
+{
+    public function __construct(PhieuChamSocRepository $phieuChamSocRepository)
+    {
+        $this->phieuChamSocRepository = $phieuChamSocRepository;
+    }
+    
+    public function createPhieuChamSoc($input)
```
Comment (thien-dang-robustaeng) [inline app/Services/PhieuChamSocService.php]: createPhieuChamSoc->create

File: app/Services/PhieuChamSocService.php [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    {
+        $this->phieuChamSocRepository->createPhieuChamSoc($input);
+    }
+    
+    public function getPhieuChamSocById($id)
```
Comment (thien-dang-robustaeng) [inline app/Services/PhieuChamSocService.php]: getById

File: app/Services/PhieuChamSocService.php [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    {
+        $data = $this->phieuChamSocRepository->getPhieuChamSocById($id);
+        return $data;
+    }
+    
+    public function getListPhieuChamSocByHsbaId($hsbaId)
```
Comment (thien-dang-robustaeng) [inline app/Services/PhieuChamSocService.php]: getListPhieuChamSocByHsbaId->getAllByHsbaId

Comment (hungbg93) [reply app/Services/PhieuChamSocService.php]: Done

File: app/Http/Controllers/Api/V1/PhieuChamSoc/PhieuChamSocController.php [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+class PhieuChamSocController extends APIController {
+    public function __construct(
+        PhieuChamSocService $phieuChamSocService
+    )
+    {
+        $this->phieuChamSocService = $phieuChamSocService;
+    }
+
+    public function createPhieuChamSoc(Request $request)
```
Comment (thien-dang-robustaeng) [inline app/Http/Controllers/Api/V1/PhieuChamSoc/PhieuChamSocController.php]: createPhieuChamSoc->create

File: app/Http/Controllers/Api/V1/PhieuChamSoc/PhieuChamSocController.php [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    {
+        $input = $request->all();
+        $this->phieuChamSocService->create($input);
+        return $this->respond([]);
+    }
+    
+    public function getPhieuChamSocById($id)
```
Comment (thien-dang-robustaeng) [inline app/Http/Controllers/Api/V1/PhieuChamSoc/PhieuChamSocController.php]: getPhieuChamSocById->getById

File: app/Http/Controllers/Api/V1/PhieuChamSoc/PhieuChamSocController.php [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    {
+        $data = $this->phieuChamSocService->getById($id);
+        return $this->respond($data);
+    } 
+    
+    public function getListPhieuChamSocByDieuTriId($dieuTriId)
```
Comment (thien-dang-robustaeng) [inline app/Http/Controllers/Api/V1/PhieuChamSoc/PhieuChamSocController.php]: getListPhieuChamSocByDieuTriId->getAllByDieuTriId

File: app/Repositories/DieuTri/DieuTriRepository.php [hunk 11]
```diff
@@ -61,5 +61,29 @@ public function getInforDieuTriById($id)
...
-    }    
+    }
+    
+    function getDetailById($id)
+    {
+        $data = $this->model->findOrFail($id);
+        return $data;
+    }
+    
+    function getListByHsbaId($hsbaId)
```
Comment (thien-dang-robustaeng) [inline app/Repositories/DieuTri/DieuTriRepository.php]: getListByHsbaId->getAllByHsbaId

File: app/Services/DieuTriService.php [hunk 12]
```diff
@@ -446,4 +446,21 @@ public function getPhieuDieuTri(array $input)
...
+        $this->dieuTriRepository->createDataDieuTri($input);
+    }
+    
+    public function getDetailById($id)
+    {
+        $data = $this->dieuTriRepository->getDetailById($id);
+        return $data;
+    }
+    
+    public function getListByHsbaId($hsbaId)
```
Comment (thien-dang-robustaeng) [inline app/Services/DieuTriService.php]: getListByHsbaId->getAllByHsbaId

File: app/Http/Controllers/Api/V1/DieuTri/DieuTriController.php [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+        return $this->respond([]);
+    }
+    
+    public function getDetailById($id)
+    {
+        $data = $this->dieuTriService->getDetailById($id);
+        return $this->respond($data);
+    } 
+    
+    public function getListByHsbaId($hsbaId)
```
Comment (thien-dang-robustaeng) [inline app/Http/Controllers/Api/V1/DieuTri/DieuTriController.php]: getListByHsbaId->getAllByHsbaId

**Pull request merged**